x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/EnrichCoordinatorProxyAction.java

```diff
@@ -66,6 +67,7 @@ protected void doExecute(Task task, SearchRequest request, ActionListener<Search
             // Write tp is expected when executing enrich processor from index / bulk api
             // Management tp is expected when executing enrich processor from ingest simulate api
             assert Thread.currentThread().getName().contains(ThreadPool.Names.WRITE)
+                || Thread.currentThread().getName().contains(ThreadPool.Names.SEARCH)
```

Comment:
Maybe add a comment here describing why we for now temporarily allow search tp here?